haystack/modeling/training/base.py

```diff
@@ -371,7 +373,24 @@ def train(self):
 
     def compute_loss(self, batch: dict, step: int) -> torch.Tensor:
         # Forward & backward pass through model
-        logits = self.model.forward(**batch)
+        if isinstance(self.model, AdaptiveModel):
+            logits = self.model.forward(
+                input_ids=batch["input_ids"], segment_ids=None, padding_mask=batch["padding_mask"]
+            )
+
+        elif isinstance(self.model, BiAdaptiveModel):
+            logits = self.model.forward(
+                query_input_ids=batch["query_input_ids"],
+                query_segment_ids=batch["query_segment_ids"],
+                query_attention_mask=batch["query_attention_mask"],
+                passage_input_ids=batch["passage_input_ids"],
+                passage_segment_ids=batch["passage_segment_ids"],
+                passage_attention_mask=batch["passage_attention_mask"],
+            )
+
+        else:
+            logits = self.model.forward(**batch)
```

Comment:
I think if it ever goes in this branch it will fail, the batch contains plenty of stuff that nobody is ready to take anymore. I'd rather raise an exception.

Reply:
Ok, interesting perspective. My perspective is - this is some model we potentially don't know about (including TriAdaptiveModel that accepts kwargs) therefore since we don't know the actual forward parameters - use all of them.

Reply:
Ah, TriAdaptive still accepts kwargs! Ok I didn't notice that. Alright let's keep it then :+1: